build-system/server.js

```diff
@@ -440,6 +440,15 @@ app.get(['/examples/*', '/test/manual/*'], function(req, res, next) {
   filePath = filePath.substr(0, filePath.length - 9) + '.html';
   fs.readFileAsync(process.cwd() + filePath, 'utf8').then(file => {
     file = replaceUrls(mode, file);
+
+    // Extract amp-ad for the given 'type' specified in URL query.
+    if (req.path.indexOf('/examples/ads.amp') == 0 && req.query.type) {
+      var ads = file.match(new RegExp('<amp-ad [^>]*'
+          + req.query.type + '[^>]*>([\\s\\S]+?)<\/amp-ad>', 'gm'));
```

Comment:
Interesting solution :-), that's a good way to think about it. But I suppose the result will be different as this will keep all other HTML tags on the page. 

And it doesn't work right now. So I don't bother to spend extra time on this, as my current solution is not huge complicated either.